src/ansys/fluent/core/session_meshing.py

```diff
@@ -0,0 +1,202 @@
+"""Module containing class encapsulating Fluent connection."""
+import grpc
+
+from ansys.fluent.core.services.datamodel_se import PyMenuGeneric
+from ansys.fluent.core.services.datamodel_tui import TUIMenuGeneric
+from ansys.fluent.core.session import (
+    _CODEGEN_MSG_TUI,
+    BaseSession,
+    parse_server_info_file,
+)
+from ansys.fluent.core.session_solver import Solver
+from ansys.fluent.core.utils.logging import LOG
+
+_CODEGEN_MSG_DATAMODEL = (
+    "Currently calling the datamodel API in a generic manner. "
+    "Please run `python codegen/allapigen.py` from the top-level pyfluent "
+    "directory to generate the local datamodel API classes."
+)
+
+
+class Meshing(BaseSession):
+    """Encapsulates a Fluent - Meshing session connection.
+    Meshing(Session) holds the top-level objects
+    for meshing TUI and various meshing datamodel API calls."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        port: int = None,
+        password: str = None,
+        channel: grpc.Channel = None,
+        cleanup_on_exit: bool = True,
+        start_transcript: bool = True,
+        remote_instance=None,
+        fluent_connection=None,
+    ):
+        super().__init__(
+            ip=ip,
+            port=port,
+            password=password,
+            channel=channel,
+            cleanup_on_exit=cleanup_on_exit,
+            start_transcript=start_transcript,
+            remote_instance=remote_instance,
+            fluent_connection=fluent_connection,
+        )
+        self._tui_service = self.fluent_connection.datamodel_service_tui
+        self._se_service = self.fluent_connection.datamodel_service_se
+        self._tui = None
+        self._meshing = None
+        self._workflow = None
+        self._part_management = None
+        self._pm_file_management = None
+
+        self.solver_switch = False
+
+    @classmethod
+    def create_from_server_info_file(
```

Comment:
@prmukherj the following can be addressed in https://github.com/pyansys/pyfluent/issues/699 also.  This will be a class method at the base class level. Here's the general pattern:

```
>>> class Base:
...   @classmethod
...   def create_instance(cls):
...     return cls()
...
>>> class Derived(Base):
...   pass
...
>>> d = Derived.create_instance()
>>>
>>> d
<__main__.Derived object at 0x000001B983B05940>

```